### ISSUE TITLE
v8.2.1 is not able to support some Japanese characters.

### ISSUE THREAD

**Issue by Michin0suke** [TarekRaafat/autoComplete.js#169]
**System Information**
  - [ ] Browser type and version : Chrome, Safari (What I have confirmed)
  - [ ] OS type and version Mac 10.15.7, iPhone 13.7
  - [ ] Node version (IF applicable) : v15.0.1
  - [ ] Any error messages in the JS console : Nothing showed up.

- [ ] **Describe the bug**
When I type some Japanese words in v8.2.1, the result were not displayed.
The problematic characters are as follows.
> ゔ
がぎぐげご
ざじずぜぞ
だぢづでど
ばびぶべぼ
ヴ
ガギグゲゴ
ザジズゼゾ
ダヂヅデド
バビブベボ
ぱぴぷぺぽ
パピプペポ

If any of the above characters are included, the result will not be displayed.

These characters are common in Japanese and are also used for particles.

When I reverted to v7.2.0, it worked fine for these characters.

Here is some example.

```js
// source
[
{name, "ひじり"}
{name, "ひかげ"}
]
```

| type | result               |
| ---- | -------------------- |
| ひ   | ["ひかげ", "ひじり"] |
| ひじ | []                   |
| ひり | ["ひじり"]           |

- [ ] **To Reproduce**
1. Type above word
2. The results disappear

- [ ] **Expected behavior**
Results containing the above characters will be displayed.

- [ ] **Screenshots or GIF's (optional, but HIGHLY RECOMMENDED)**
![autoCompleteJs](https://user-images.githubusercontent.com/49580786/104783124-b82d8a00-57c8-11eb-885a-386cf7f31939.gif)

**Comment by TarekRaafat** [comment]
Hello @Michin0suke,

Thanks for reporting this issue.

I'll investigate it and get back to you asap.

Have you tried turning on the `diacritics` API?


Have a nice day, cheers! :)

**Comment by Michin0suke** [comment]
Hello @TarekRaafat.

Yes.
`diacritics` API is set to `false` and `searchEngine` is set to `loose`.
Also, I have already confirmed that the problem does not occur in v7.2.0 with the exact same settings.

* Node v15.0.1 + v7.2.0 → OK
* Node v15.0.1 + v8.2.1 → Does not work

This is a fatal problem for Japanese people, so please solve the problem.

**Comment by bravik** [comment]
Hi, same for Russian. The character `й` will cause the results to be ignored.
I do not really need highlighting, so I've just turned off `searchEngine` option  to get missing suggestions:
```
searchEngine: (query, record) => record,
```

**Comment by ikemo3** [comment]
The reason it doesn't work with the default settings is that the input is NFD-normalized, but the data is not NFD-normalized.
If `diacritics: true` is set, autocomplete will work.

https://github.com/TarekRaafat/autoComplete.js/blob/43aa7068e849e272e7f09024b777ec99d32673dc/dist/js/autoComplete.js#L280
https://github.com/TarekRaafat/autoComplete.js/blob/43aa7068e849e272e7f09024b777ec99d32673dc/src/controllers/dataController.js#L22-L26

**Comment by ikemo3** [comment]
The issue in Japanese was solved in #171.

@bravik 

> The character `й` will cause the results to be ignored.

`й` has a diacritic mark, so I think it works with `diacritics: true` (it worked in my environment).

**Comment by TarekRaafat** [comment]
Thanks to you all guys specially @ikemo3 for solved this issue #171.

Here is the new v8.2.2 on [Github](https://github.com/TarekRaafat/autoComplete.js/releases/tag/v8.2.2) and [JSDeliver](https://www.jsdelivr.com/package/gh/TarekRaafat/autoComplete.js)
